### PR TITLE
Fix 21952 - Don't assert for global / tls variables of type noreturn

### PR DESCRIPTION
**Pull request by MoonlightSentinel**
The old code assumed variables of size 0 must be static arrays `T[0]` but `noreturn.sizeof` is 0.

### CODE REVIEW

**Review by thewilsonator — APPROVED** [review]

File: test/compilable/noreturn1.d [hunk 1]
```diff
@@ -85,3 +85,20 @@ int test1(int i)
         return i + 1;
     return i - 1;
 }
+
+noreturn tlsNoreturn;
+__gshared noreturn globalNoreturn;
```
Comment (ibuclaw) [inline test/compilable/noreturn1.d:90]: This is messed up what is even the default initializer of these types?  As far as I can tell its just `assert(0)`, but that's not valid for static storage?

Comment (MoonlightSentinel) [reply test/compilable/noreturn1.d:90]: `noreturn` variables without initializers are allowed, the `assert(0)` is raised when accessing the variable.
DMD should only reject `noreturn` initializers for global / static variables.

See [DIP 1034, Declarations](https://github.com/dlang/DIPs/blob/master/DIPs/accepted/DIP1034.md#interaction-with-other-language-features)

Comment (ibuclaw) [reply test/compilable/noreturn1.d:90]: Without initializers means `= void` though. ;-)

Comment (MoonlightSentinel) [reply test/compilable/noreturn1.d:90]: Usually yes. But the DIP explicitly refers to variables without an initialization expression, i.e. neither value nor `void`.

> Defining a variable with type noreturn with an initialization expression will simply generate that expression as soon as it becomes live. Defining a noreturn variable with no initialization expression generates an assert(0) only if the variable is accessed, [...]

**Pull request merged**